test/connection.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-env node, mocha */
+require('./init.js');
+var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+
+var config;
+
+before(function() {
+  config = global.getConfig();
+  console.log("config", config);
+});
+
+describe('testConnection', function() {
+  it('should pass with valid settings', function(done) {
+    var db = new DataSource(require('../'), config);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
```

Comment:
In fact, when I read the code agian. You should just pass done to ping as @bajtos stated. `db.ping(done);`